agent/scheduler.py

```
@@ -204,8 +204,18 @@ def delete(self, job_id: str) -> bool:
+        running = self._running_loop()
+        if running is not None:
             asyncio.create_task(self._remove_persisted(job_id))
+        elif self._main_loop is not None:
+            # APScheduler thread: dispatch persistence delete onto the FastAPI
+            # main loop so it can safely reach Motor/aiosqlite clients.
+            try:
+                asyncio.run_coroutine_threadsafe(
+                    self._remove_persisted(job_id), self._main_loop,
+                )
+            except Exception:
+                pass
```

Ruff (0.15.18)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)